tools/schemacode/bidsschematools/schema.py

```diff
@@ -274,3 +284,21 @@ def filter_schema(schema, **kwargs):
             if isinstance(item, dict):
                 new_schema[i] = filter_schema(item, **kwargs)
     return new_schema
+
+
+def validate_schema(schema: Namespace):
+    """Validate a schema against the BIDS metaschema."""
+    metaschema = json.loads(files("bidsschematools.data").joinpath("metaschema.json").read_text())
+
+    # validate is put in this try/except clause because the error is sometimes too long to
+    # print in the terminal
+    try:
+        validate(instance=schema.to_dict(), schema=metaschema)
+    except ValidationError as e:
+        with tempfile.NamedTemporaryFile(
+            prefix="schema_error_", suffix=".txt", delete=False, mode="w+"
+        ) as file:
+            file.write(str(e))
+            # ValidationError does not have an add_note method yet
+            # e.add_note(f"See {file.name} for full error log.")
```

Comment:
Apparently this is a 3.11+ builtin: 

https://docs.python.org/3/library/exceptions.html#BaseException.add_note

It would be nice to display. Not sure if there's a way to hack this until 3.11 is our minimum.